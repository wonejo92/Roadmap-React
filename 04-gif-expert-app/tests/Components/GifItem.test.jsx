import {GifItem} from "../../src/Components/index.js";
import {render, screen} from "@testing-library/react";

describe('GifItem Test', () => {
    const title = "GifItem";
    const url = "https://api.giphy.com/v1/gifs/";
    test('should render correctly and match with the snapshot', () => {
        const{container}=render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })
    test('should show the picture with the url correctly like the alt', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        const{src,alt}=screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })
    test('should show the title on the component', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})