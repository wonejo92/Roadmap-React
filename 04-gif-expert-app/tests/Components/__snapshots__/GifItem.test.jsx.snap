// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Test should render correctly and match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GifItem"
      src="https://api.giphy.com/v1/gifs/"
    />
    <p>
      GifItem
    </p>
  </div>
</div>
`;
